.travis.yml

```
@@ -64,12 +64,19 @@ before_install:
         MULTIPASS_REPORT_OPTIONS+=("--branch" "${TRAVIS_BRANCH}")
       fi
 
-    # other branches only publish on the commits
+    # other branches publish on the commits by default
     # skip staging as the same commit will be published from master if all is good
     elif [[ "${TRAVIS_EVENT_TYPE}" == "push" \
             && "${TRAVIS_BRANCH}" != "staging" ]]; then
       MULTIPASS_PUBLISH="true"
 
+      # except for `trying`, which publishes on the parent's PR
+      if [[ "${TRAVIS_BRANCH}" == "trying" \
+            && "${TRAVIS_COMMIT_MESSAGE}" =~ ^Try\ #([0-9]+): ]]; then
```

@ricab one thing I realized I forgot:

```suggestion
        MULTIPASS_SNAP_CHANNEL="edge/pr${BASH_REMATCH[1]}"
```

This makes the channel different from the build label, but consistent in terms of snap channels, which I think is more important. I could also just keep the build label `pr*`, so it's consistent regardless if it's a PR build or a `trying` one… WDYT?

---

Hmm so without this, a trying snap wouldn't be published in snapcraft, right? Do we really need it there? With this, it would overwrite the snap in the corresponding pr channel, correct?

If so, I think I'd prefer a separate `edge/tryXYZ` channel, to keep the distinction transparent and consistent with the build label. If there is any difference between that and the previous PR build, better be able to investigate it. Otherwise, what's the real point of publishing? 

I recognize the proliferation of snap channels, but I think that is fair reflex of the many sorts of build uploads.

---

@ricab we only really need `bors try` on external contributions, on which the PR builds won't get published in the store because of the lack of credentials.

---

Oh and the only difference between a PR build and a `bors try` build would be the merge commit, so in the package - the version string. And even if a channel is "overwritten", collaborators have access to previous revisions via `snap install --revision`.

The argument for keeping it under the `edge/pr*` branch is to avoid reviewers having to check / remember. The most recent build for a PR, whether from a `pull_request` travis build, or a `bors try` one, would be published in the `edge/pr<number>` channel.

Maybe the `try` build label is an unnecessary distinction?

---

> I am not sure I follow. We use `bors try` ourselves sometimes. But if the only use would fail uploading to the store, why attempt to upload in the first place?

It's exactly `bors try` that won't fail to upload, as opposed to a `pull_request` build from outside of the CanonicalLtd/multipass repository (for security reasons - we can't run untrusted code with credentials). `bors try` means that someone has looked at the code and verified there's nothing malicious in it and it's safe to run with credentials, we can very well publish for testing. When the PR is "internal", `bors try` does virtually the same thing a normal PR run does.

> OK, but then what is the use of publishing it?

Testing, no?

---

Ah ok, much clearer now. 

Well, if there is no reason to distinguish snap channels, indeed I'd expect no real reason to distinguish snap package names. So yeah, IMO it should be coherent: either `try*` or `pr*` in both label and channel.